client/extensions/woocommerce/state/sites/products/reducer.js

```diff
@@ -81,15 +88,12 @@ export function productsDeleteSuccess( state, action ) {
 	};
 }
 
-export function productsRequest( state, action ) {
-	const prevState = state || {};
-	const isLoading = setLoading( prevState, action.page, true );
-	return { ...prevState, isLoading };
-}
-
-export function productsRequestFailure( state, action ) {
+export function productsRequest( state, { page, meta: { dataLayer: { error, data } } } ) {
+	if ( data || error ) {
```

Comment:
I'm not certain this is what we should be looking for. In fact, from my understanding of the data-layer, this action should only get passed to the reducers once, the first time. After that it shouldn't be dispatched down further than the data-layer handlers. It kind of defeats the purpose of simplifying actions in the data layer if we have to guard for meta values here.

What that may mean is that we need to remove `next( action );` from the handler code to ensure that the action doesn't make it down to the reducer and only the `RECEIVE` action does.